### PR TITLE
Add --watch to test app

### PR DESCRIPTION
**Pull request by wolfs**
So we can easily try out watching a directory on macOs or Windows.

### CODE REVIEW

File: test-app/src/main/java/net/rubygrapefruit/platform/test/Main.java [hunk 1]
```diff
@@ -347,6 +372,49 @@ private static void fileSystems() {
...
+        }
+        try {
+            System.out.println("Waiting - type ctrl-d to exit ...");
+            while (true) {
+                int ch = System.in.read();
+                if (ch < 0) {
+                    break;
+                }
+            }
+            watcher.close();
```
Comment (lptr) [inline test-app/src/main/java/net/rubygrapefruit/platform/test/Main.java]: Why close twice?

Comment (wolfs) [reply test-app/src/main/java/net/rubygrapefruit/platform/test/Main.java]: Closing only once now.

**Comment by lptr** [comment]
Now that https://github.com/adammurdoch/native-platform/pull/58 is merged we can remove the `System.exit()`.

**Comment by wolfs** [comment]
I removed the `System.exit()` call.

File: test-app/src/main/java/net/rubygrapefruit/platform/test/Main.java [hunk 2]
```diff
@@ -19,21 +19,40 @@
...
+import net.rubygrapefruit.platform.file.FileSystemInfo;
+import net.rubygrapefruit.platform.file.FileSystems;
+import net.rubygrapefruit.platform.file.FileWatcher;
+import net.rubygrapefruit.platform.file.FileWatcherCallback;
+import net.rubygrapefruit.platform.file.Files;
+import net.rubygrapefruit.platform.file.PosixFileInfo;
+import net.rubygrapefruit.platform.file.PosixFiles;
+import net.rubygrapefruit.platform.internal.Platform;
+import net.rubygrapefruit.platform.internal.jni.OsxFileEventFunctions;
+import net.rubygrapefruit.platform.internal.jni.WindowsFileEventFunctions;
```
Comment (adammurdoch) [inline test-app/src/main/java/net/rubygrapefruit/platform/test/Main.java]: I don't think we should add this until there is a public API for file events, as this is also a sample and documentation for the APIs.

Comment (wolfs) [reply test-app/src/main/java/net/rubygrapefruit/platform/test/Main.java]: So you think we should remove it from `master` again for the time being? It was very helpful to use the test-app to debug problems with file watching.

Comment (adammurdoch) [reply test-app/src/main/java/net/rubygrapefruit/platform/test/Main.java]: That's one option. The other is to add a public API, which is what I would do. We just need to extract an interface or two out of the implementations.

**Pull request merged**